test/common/http/conn_manager_utility_test.cc

```diff
@@ -36,6 +36,25 @@ class ConnectionManagerUtilityTest : public testing::Test {
     ON_CALL(config_, tracingConfig()).WillByDefault(Return(&tracing_config_));
   }
 
+  struct MutateRequestRet {
+    bool operator==(const MutateRequestRet& rhs) const {
+      return downstream_address_ == rhs.downstream_address_ && internal_ == rhs.internal_;
+    }
+
+    std::string downstream_address_;
+    bool internal_;
+  };
+
+  MutateRequestRet callMutateRequestHeaders(HeaderMap& headers, Protocol protocol) {
```

Comment:
I think this results in harder to read tests than before - checking IP on return and checking the headers after return seems cleaner.

If you think it's worth the decrease in readability to force "internal" checking (reasonable, since it's scary fragile code) please add a comment.

Reply:
I did it to force it as you speculated, but if you think it makes it harder to read, I can revert to the old way. Do you feel strongly about it?

Reply:
Nope, hence a comment being fine.  As is I could imagine cleaning it up "to make it easier to read" and losing the forcing function.